File: Framework.Tests/CustomerServices/Customer.cs
using System.Collections.Generic;
using Framework.Data_Manipulation;
using Framework.Domain;
using Framework.Tests.Mapping;
using Framework.Tests.ReceivableServices;

namespace Framework.Tests.CustomerServices
{
    public class Customer : DomainObject
    {
        FK_Customer_AccountReceivable _fk_Customer_AccountReceivable = new FK_Customer_AccountReceivable();

        public Customer(IBaseMapper mapper) : base(mapper)
        {
        }

        [IdentityField]
        public string Number { get; set; }

        public string Name { get; set; }

        public IList<AccountReceivable> AccountReceivables
        {
            get { return _fk_Customer_AccountReceivable.GetChildEntities(this); }
        }
    }
}

File: Android/app/src/test/java/com/peaa/CustomerServices/Customer.java
package com.peaa.CustomerServices;

import com.peaa.IdentityFieldAnnotation;
import com.peaa.DataManipulation.BaseMapperInterfaces.IBaseMapper;
import com.peaa.Domain.DomainObject;

/**
 * Created by aiko on 3/5/17.
 */

public class Customer extends DomainObject {
    public Customer(IBaseMapper mapper)
    {
        super(mapper);
    }

    public Customer(IBaseMapper mapper, String number)
    {
        this(mapper);

        Number = number;
    }

    @IdentityFieldAnnotation
    public String Number;

    public String Name;
}
